metadb/server.go

```diff
@@ -0,0 +1,337 @@
+	// This can be started now since it doesn't require a functioning Store yet.
+	cs, err := s.startClusterService(mux.Listen(cluster.MuxClusterHeader))
+	if err != nil {
+		return errors.Errorf("start create cluster service: %v", err)
+	}
+
+	// create and open the store
+	dbConf := store.NewDBConfig(s.config.DNS, !s.config.OnDisk)
+	db := store.New(raftTn, &store.Config{
+		DBConf: dbConf,
+		Dir:    s.config.DataDir,
+		ID:     s.idOrRaftAddr(),
+	})
+
+	// set optional parameters on store
+	db.SetRequestCompression(s.config.CompressionBatch, s.config.CompressionSize)
+	db.RaftLogLevel = s.config.RaftLogLevel
+	db.ShutdownOnRemove = s.config.RaftShutdownOnRemove
+	db.SnapshotThreshold = s.config.RaftSnapThreshold
+	db.SnapshotInterval, err = time.ParseDuration(s.config.RaftSnapInterval)
+	if err != nil {
+		return errors.Errorf("parse RaftSnapInterval: %v", err)
+	}
+	db.LeaderLeaseTimeout, err = time.ParseDuration(s.config.RaftLeaderLeaseTimeout)
+	if err != nil {
+		return errors.Errorf("parse RaftLeaderLeaseTimeout: %v", err)
+	}
+	db.HeartbeatTimeout, err = time.ParseDuration(s.config.RaftHeartbeatTimeout)
+	if err != nil {
+		return errors.Errorf("parse RaftHeartbeatTimeout: %v", err)
+	}
+	db.ElectionTimeout, err = time.ParseDuration(s.config.RaftElectionTimeout)
+	if err != nil {
+		return errors.Errorf("parse RaftElectionTimeout: %v", err)
+	}
+	db.ApplyTimeout, err = time.ParseDuration(s.config.RaftApplyTimeout)
+	if err != nil {
+		return errors.Errorf("parse RaftApplyTimeout: %v", err)
+	}
+
+	// a pre-existing node
+	bootstrap := false
+	isNew := store.IsNewNode(s.config.DataDir)
+	if isNew {
+		bootstrap = true // new node, it needs to bootstrap
+	} else {
+		log.WithContext(ctx).Infof("node is detected in: %v", s.config.DataDir)
+	}
+
+	// determine the join addresses
+	joins, err := s.determineJoinAddresses(ctx)
+	if err != nil {
+		return errors.Errorf("determine join addresses: %v", err)
+	}
+	// supplying join addresses means bootstrapping a new cluster won't be required.
+	if len(joins) > 0 {
+		bootstrap = false
+		log.WithContext(ctx).Info("join addresses specified, node is not bootstrap")
+	} else {
+		log.WithContext(ctx).Info("no join addresses")
+	}
+	// join address supplied, but we don't need them
+	if !isNew && len(joins) > 0 {
+		log.WithContext(ctx).Info("node is already member of cluster")
+	}
+
+	// open store
+	if err := db.Open(bootstrap); err != nil {
+		return errors.Errorf("open store: %v", err)
+	}
+	s.db = db
+
+	// execute any requested join operation
+	if len(joins) > 0 && isNew {
+		log.WithContext(ctx).Infof("join addresses are: %v", joins)
+		advAddr := s.config.RaftAddress
+		if s.config.RaftAdvertiseAddress != "" {
+			advAddr = s.config.RaftAdvertiseAddress
+		}
+
+		// try to parse the join duration
+		joinDuration, err := time.ParseDuration(s.config.JoinInterval)
+		if err != nil {
+			return errors.Errorf("parse JoinInterval: %v", err)
```

Comment:
```suggestion
			return errors.Errorf("parse JoinInterval: %w", err)
```
always use `%w` with errors, to not lose already the wrapped errors

Reply:
done